src/python/dxpy/bindings/timeout.py

```diff
@@ -0,0 +1,27 @@
+# https://stackoverflow.com/questions/2281850/timeout-function-if-it-takes-too-long-to-finish
+
+from functools import wraps
+import errno
+import os
+import signal
+
+class TimeoutError(Exception):
+    pass
+
+def timeout(seconds=10, error_message=os.strerror(errno.ETIME)):
```

Comment:
According to https://docs.python.org/2/library/signal.html#signal.signal SIGALARM is not available for Windows:

```
On Windows, signal() can only be called with SIGABRT, SIGFPE, SIGILL, SIGINT, SIGSEGV,
SIGTERM, or SIGBREAK. A ValueError will be raised in any other case 
```

so we could catch it and simply return the function there (without the timeout handler).

According to https://stackoverflow.com/questions/8420422/python-windows-equivalent-of-sigalrm, it can be made work if python is installed with Cygwin on Windows, I’m not sure how we use/install Python there.

Reply:
Done